Cargo.toml

```diff
@@ -67,6 +67,7 @@ duckdb = "1.0.0"
 fundu = "2.0.0"
 futures = "0.3.30"
 globset = "0.4.14"
+graph-rs-sdk = "2.0.1"
```

Comment:
Might be good to add feature flag for Sharepoint connector as it brings new dependencies